suite/tests/linux/syscall_pwait.cpp

```diff
@@ -433,7 +437,7 @@ main(int argc, char *argv[])
         } data;
         data.sigmask = NULL;
         data.ss_len = 0;
-        return syscall(SYS_pselect6, 0, 0, 0, &fds, &ts, &data);
+        return syscall(SYS_pselect6, NULL, NULL, NULL, &fds, &ts, &data);
```

Comment:
I understand making the 2nd and 3rd NULL b/c they're pointers, but the first is an integer, so 0 is better.  See the comment above: is this b/c you're writing into the %r instead of %e register??

Reply:
Fixed. I went through every single parameter in the test and am now casting into their correct types according to their function declaration. Even NULL might actually not be 100% safe, thought the compiler might conservatively assume that NULL may be a LL pointer (but could be an int). Only the pointers are being passed a casted NULL now, and the rest is passing literals. This should be ultra-safe now.

Reply:
You made this a C++ test, so `nullptr` should resolve worries about the size of `NULL`.